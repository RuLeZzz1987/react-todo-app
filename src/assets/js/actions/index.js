export CategoryActions from './CategoryActions';
export TodoActions from './TodoActions';
export FilterActions from './FilterActions';