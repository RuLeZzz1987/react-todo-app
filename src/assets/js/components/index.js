import React, { Component } from 'react';
import Header from './header';
import Main from './main';
import { Category, Todo } from '../models';

class App extends Component {
    
    constructor(props) {
        super(props);
        
        this.state = {
            categories: [new Category({name: 'Category_1', isRoot: true, children: [new Todo({name: 'To-Do Item #1'}), new Category({name: 'Category_1_1'})]})]
        };
        
        this.updateCategories = (categories, cb) => this.setState({categories}, cb);
       
    }
    
    render() {
        return (
            <div>
                <Header
                    categories={this.state.categories}
                />
                <Main
                    categories={this.state.categories}
                    updateCategories={this.updateCategories}
                />
            </div>
        )
    }
}

export default App;