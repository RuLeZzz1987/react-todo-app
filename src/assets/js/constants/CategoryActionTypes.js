const Types = {
  ADD_CATEGORY: Symbol('ADD_CATEGORY'),
  EDIT_CATEGORY: Symbol('EDIT_CATEGORY'),
  REMOVE_CATEGORY: Symbol('REMOVE_CATEGORY'),
  TOGGLE_CATEGORY: Symbol('TOGGLE_CATEGORY'),
};

export default Types