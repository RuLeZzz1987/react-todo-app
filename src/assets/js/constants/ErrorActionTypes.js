const Types = {
  'SET_ERROR': Symbol('SET_ERROR'),
  'CLEAR_ERROR': Symbol('CLEAR_ERROR'),
  'SHOW_IN_POPUP': Symbol('SHOW_IN_POPUP'),
  'SET_ERROR_TYPE': Symbol('SET_ERROR_TYPE'),
};

export default Types