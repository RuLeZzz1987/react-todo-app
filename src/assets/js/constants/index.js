
export const CATEGORY = Symbol('category');

export const TODO = Symbol('todo');