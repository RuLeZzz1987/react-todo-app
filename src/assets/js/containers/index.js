export Header from './HeaderContainer';
export Categories from './CategoriesContainer';
export Category from './CategoryContainer';
export Editor from './EditorContainer';
export ModalError from './ModalErrorContainer';