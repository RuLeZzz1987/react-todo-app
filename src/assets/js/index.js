import React from 'react';
import { render } from 'react-dom';
import App from './components'
import '../styles/styles.scss';

render(<App/>, document.querySelector('#root'));