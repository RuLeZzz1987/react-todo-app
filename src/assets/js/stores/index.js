export TodoStore from './TodoStore';
export CategoryStore from './CategoryStore';
export FilterStore from './FilterStore';
export ErrorStore from './ErrorStore';