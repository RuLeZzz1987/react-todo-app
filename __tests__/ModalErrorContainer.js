import React from 'react';
import renderer from 'react-test-renderer';
import { ModalError } from '../src/assets/js/containers';
import { TODO } from '../src/assets/js/constants';
import { ErrorStore } from '../src/assets/js/stores';

describe('ModalErrorContainer', function () {

  beforeEach(function () {

    ErrorStore._state = ErrorStore.getInitialState();

    this.render = () => renderer.create(
      <ModalError />
    ).toJSON();

  });

  it('can render hidden modal without error', function () {

    expect(this.render()).toMatchSnapshot();

  });

  it('can render shown modal with error message', function () {

    ErrorStore._state = {
      isError: true,
      showInPopup: true,
      message: 'error in popup',
      type: TODO
    };

    expect(this.render()).toMatchSnapshot();

  });

  it('can render hidden modal with error message', function () {

    ErrorStore._state = {
      isError: true,
      showInPopup: false,
      message: 'error in popup',
      type: TODO
    };

    expect(this.render()).toMatchSnapshot();

  });

});