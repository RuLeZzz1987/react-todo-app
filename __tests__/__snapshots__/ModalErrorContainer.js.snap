exports[`ModalErrorContainer can render hidden modal with error message 1`] = `
<div
  className="modal"
  onClick={[Function]}
  style={
    Object {
      "display": "none",
    }
  }>
  <div
    className="modal-content">
    <div
      className="modal-header">
      <span
        className="close"
        onClick={[Function]}>
        ×
      </span>
      <h2>
        Something goes wrong!
      </h2>
    </div>
    <div
      className="modal-body">
      <p>
        error in popup
      </p>
    </div>
  </div>
</div>
`;

exports[`ModalErrorContainer can render hidden modal without error 1`] = `
<div
  className="modal"
  onClick={[Function]}
  style={
    Object {
      "display": "none",
    }
  }>
  <div
    className="modal-content">
    <div
      className="modal-header">
      <span
        className="close"
        onClick={[Function]}>
        ×
      </span>
      <h2>
        Something goes wrong!
      </h2>
    </div>
    <div
      className="modal-body">
      <p>
        
      </p>
    </div>
  </div>
</div>
`;

exports[`ModalErrorContainer can render shown modal with error message 1`] = `
<div
  className="modal"
  onClick={[Function]}
  style={
    Object {
      "display": "block",
    }
  }>
  <div
    className="modal-content">
    <div
      className="modal-header">
      <span
        className="close"
        onClick={[Function]}>
        ×
      </span>
      <h2>
        Something goes wrong!
      </h2>
    </div>
    <div
      className="modal-body">
      <p>
        error in popup
      </p>
    </div>
  </div>
</div>
`;
